build/settings/llvm-release.mk

```diff
@@ -14,6 +14,10 @@ export CFLAGS   := -Os -g -gcodeview -fdata-sections -ffunction-sections
 export CXXFLAGS := -Os -g -gcodeview -fdata-sections -ffunction-sections
 export LDFLAGS  := -Wl,--pdb= -Wl,--gc-sections -Wl,-s
 
+# Force inclusion of a couple of GLib symbols that would otherwise
+# get removed by --gc-sections.
+export LDFLAGS += -Wl,-u,g_atomic_int_inc -Wl,-u,g_atomic_int_dec_and_test
```

Comment:
This is a bit unfortunate. Without it, linking fails with:
```
metadata.obj : error LNK2001: unresolved external symbol __imp_g_atomic_int_inc [C:\Users\kleisauke\sharp\build\sharp-win32-x64.vcxproj]
metadata.obj : error LNK2001: unresolved external symbol __imp_g_atomic_int_dec_and_test [C:\Users\kleisauke\sharp\build\sharp-win32-x64.vcxproj]
C:\Users\kleisauke\sharp\build\Release\sharp-win32-x64.node : fatal error LNK1120: 2 unresolved externals [C:\Users\kleisauke\sharp\build\sharp-win32-x64.vcxproj]
```

Reply:
I could refactor sharp to use something else instead if that helps? [Napi::TypedThreadSafeFunction](https://github.com/nodejs/node-addon-api/blob/main/doc/typed_threadsafe_function.md) looks like it might be suitable, or maybe just do the whole thing in JS land?

Reply:
I'm not sure what's the ideal solution for this. I think the issue is that `g_atomic_int_inc()` and `g_atomic_int_dec_and_test()` are implemented as macro when compiling with GCC or Clang.
https://github.com/GNOME/glib/blob/2.78.0/glib/gatomic.h#L161-L172

Whereas MSVC forwards them to their corresponding implementations.
https://github.com/GNOME/glib/blob/2.78.0/glib/gatomic.h#L547-L550
https://github.com/GNOME/glib/blob/2.78.0/glib/gatomic.c#L710-L720

So, when you use `--gc-sections`, the linker thinks that the implementations of both `g_atomic_int_inc()` and `g_atomic_int_dec_and_test()` aren't needed and removes it as part of DCE (dead-code elimination).

Reply:
Swapping these calls for `std::atomic` should work - I'll do that and then you can remove them from this PR.

Reply:
Removed via commit https://github.com/lovell/sharp/commit/70ac6905c74f64de02fc956b26be2e8ea356c207

Reply:
Great! Commit 10d1585261a5c05ad73ea44ca2857c15c651a70a removes this workaround.